cores/esp8266/core_esp8266_main.cpp

```diff
@@ -145,6 +145,7 @@ void init_done() {
     system_set_os_print(1);
     gdb_init();
     do_global_ctors();
+    early_setup(); // <-- here after global constructors, or in user_init() ?
```

Comment:
Bottom of user_init(), not here.  This is called after WiFi power probing/etc. is set up.

Reply:
fixed